internal/shared/db/client/locks.go

```diff
@@ -0,0 +1,10 @@
+package dbclient
+
+import (
+	dbmodel "github.com/babylonlabs-io/staking-api-service/internal/shared/db/model"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func (db *Database) LocksCollection() *mongo.Collection {
```

Comment:
that's a little bit hacky, but I didn't want to put here locks logic cause to me it's not db specific, on the other hand I return `mongo.Collection` here which is usually abstracted inside db classes, wdyt?